src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -174,13 +174,8 @@ public StatementRestrictions(StatementType type,
             }
             else if (relation.isLIKE())
             {
+                // CASSANDRA-17198 Remove restrictions in LIKE statement.
                 Restriction restriction = relation.toRestriction(table, boundNames);
-
-                if (!type.allowUseOfSecondaryIndices() || !restriction.hasSupportingIndex(indexRegistry))
-                    throw new InvalidRequestException(String.format("LIKE restriction is only supported on properly " +
-                                                                    "indexed columns. %s is not valid.",
-                                                                    relation.toString()));
-
```

Comment:
You can remove that if clause as it is now similar to the next one

Reply:
Removed.